### ISSUE TITLE
ResourceOutputStream treats open stream as closed

### ISSUE THREAD

**Issue by psafarov** [amphp/byte-stream#58]
Here is a problem which occurs if the script is run from systemd
```php
Amp\Loop::run(function() {
    $resourceOutputStream = new ByteStream\ResourceOutputStream(\STDOUT);
    // Throws Amp\ByteStream\StreamException: The stream was closed by the peer
    yield $resourceOutputStream->write('test');
});
```
I figured out that it stops happening if `feof` check of the stream is removed in `ResourceOutputStream.php:74`. Apparently it is not correct to use `feof` for checking if stream is open or not, but I am not sure what can be done here as it has been added for a reason.

**Comment by bwoebi** [comment]
No idea why feof() is being used, this only makes sense for readability checks - it has no purpose in ResourceOutputStream.

**Comment by trowski** [comment]
@bwoebi I think what really fixed #40 was failing after multiple writes of 0 length. On macOS `feof()` detects IPC pipes being closed, but I always had mixed results with network sockets. If it causes trouble on other systems, we can probably drop the two if-statements with `feof()` and rely on failing after multiple 0-length writes.

**Comment by kelunik** [comment]
Failing after multiple 0-length writes still isn't the optimal solution, but might be OK given we don't really support TLS renegotiation.

**Comment by kelunik** [comment]
Running the example via the shell yields the following output:

```
array(9) {
  ["timed_out"]=>
  bool(false)
  ["blocked"]=>
  bool(true)
  ["eof"]=>
  bool(false)
  ["wrapper_type"]=>
  string(3) "PHP"
  ["stream_type"]=>
  string(5) "STDIO"
  ["mode"]=>
  string(2) "wb"
  ["unread_bytes"]=>
  int(0)
  ["seekable"]=>
  bool(true)
  ["uri"]=>
  string(12) "php://stdout"
}
test
```

Running the same script with systemd yields:

```
array(9) {
  ["timed_out"]=>
  bool(false)
  ["blocked"]=>
  bool(true)
  ["eof"]=>
  bool(false)
  ["wrapper_type"]=>
  string(3) "PHP"
  ["stream_type"]=>
  string(10) "tcp_socket"
  ["mode"]=>
  string(2) "r+"
  ["unread_bytes"]=>
  int(0)
  ["seekable"]=>
  bool(false)
  ["uri"]=>
  string(12) "php://stdout"
}
PHP Fatal error:  Uncaught Amp\ByteStream\StreamException: The stream was closed by the peer
```

**Comment by kelunik** [comment]
Finally released v1.6.0. :tada:
